contracts/AddrArrayLib.sol

```diff
@@ -0,0 +1,88 @@
+/*
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity >=0.4.25 <0.6.0;
+
+library AddrArrayLib {
+    using AddrArrayLib for Addresses;
+
+    struct Addresses {
+      address[]  _items;
+    }
+
+    /**
+     * @notice push an address to the array
+     * @dev if the address already exists, it will not be added again
+     * @param self Storage array containing address type variables
+     * @param element the element to add in the array
+     */
+    function pushAddress(Addresses storage self, address element) internal {
+      if (!exists(self, element)) {
+        self._items.push(element);
+      }
+    }
+
+    /**
+     * @notice remove an address from the array
+     * @dev it wil find the element and then Swap it with the last element and then delete it

```

Comment:
it wil find the element and then Swap it with the last element and then delete it -> 
finds the element, swaps it with the last element, and then deletes it; returns a boolean whether the element was found and deleted